client/landing/stepper/declarative-flow/internals/hooks/use-flow-navigation/index.tsx

```diff
@@ -0,0 +1,79 @@
+import { OnboardSelect } from '@automattic/data-stores';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { generatePath, useMatch, useNavigate } from 'react-router';
+import { useSearchParams } from 'react-router-dom';
+import { ONBOARD_STORE, STEPPER_INTERNAL_STORE } from 'calypso/landing/stepper/stores';
+import { Navigate, StepperStep } from '../../types';
+
+const useOnboardingIntent = () => {
+	const intent = useSelect(
+		( select ) => ( select( ONBOARD_STORE ) as OnboardSelect ).getIntent(),
+		[]
+	);
+	return intent;
+};
+
+const addQueryParams = ( uri: string, params?: URLSearchParams | null ) => {
+	if ( params ) {
+		return uri + '?' + params.toString();
```

Comment:
What if the URL has query parameters already?

Reply:
I respected the [original behavior](https://github.com/Automattic/wp-calypso/pull/91935/files#diff-1a5cb229a76dfccdc8b471efedc2ab09f2385c9ba00ae3211ba65452ba1c7680L102-L105) as described in these [tests](https://github.com/Automattic/wp-calypso/blob/aef0abfe8e97a0de922621a079f1374562d3ce23/client/landing/stepper/declarative-flow/internals/hooks/use-flow-navigation/test/index.tsx#L84-L101)

The original behavior is: 
- New URL has queryParams, use them and ignore the existing ones.
- New URL doesn't have queryParams, use the existent ones.